### PR TITLE
Show idle status in `loginctl list-sessions`

### PR DESCRIPTION
**Pull request by dtardon**

### CODE REVIEW

**Review by yuwata — APPROVED** [review]
LGTM.
We should really use `bus_map_all_properties()`, but let's do that later.

**Review by yuwata — APPROVED** [review]

**Comment by dtardon** [comment]
One thing I haven't thought about earlier: what should happen when `bus_map_all_properties()` fails? The current code logs a warning and continues, printing potentially wrong values, which is probably not the right thing to do. Maybe it would be better to exit early. Or make the code a bit more complicated and print `n/a` for the properties.

File: src/login/loginctl.c [hunk 1]
```diff
@@ -143,57 +171,34 @@ static int list_sessions(int argc, char *argv[], void *userdata) {
...
-                                               &tty);
+                r = bus_map_all_properties(bus, "org.freedesktop.login1", object, map, BUS_MAP_BOOLEAN_AS_BOOL, &e, &m, &i);
                 if (r < 0) {
                         if (sd_bus_error_has_name(&e, SD_BUS_ERROR_UNKNOWN_OBJECT))
                                 /* The session is already closed when we're querying the property */
                                 continue;
 
-                        log_warning_errno(r, "Failed to get TTY for session %s, ignoring: %s",
+                        log_warning_errno(r, "Failed to get properties of session %s, ignoring: %s",
                                           id, bus_error_message(&e, r));
```
Comment (yuwata) [inline src/login/loginctl.c]: Ah, indeed. On error, please skip the session. How about the following?
```c
if (r < 0) {
        log_full_errno(sd_bus_error_has_name(&e, SD_BUS_ERROR_UNKNOWN_OBJECT) ? LOG_DEBUG : LOG_WARNING, r, ...);
        continue;
}
```

**Review by yuwata — CHANGES_REQUESTED** [review]

File: src/login/loginctl.c (hunk 1 above)
Comment (dtardon) [reply src/login/loginctl.c]: Done.

File: src/login/loginctl.c [hunk 2]
```diff
@@ -184,12 +188,12 @@ static int list_sessions(int argc, char *argv[], void *userdata) {
...
                 r = bus_map_all_properties(bus, "org.freedesktop.login1", object, map, BUS_MAP_BOOLEAN_AS_BOOL, &e, &m, &i);
                 if (r < 0) {
-                        if (sd_bus_error_has_name(&e, SD_BUS_ERROR_UNKNOWN_OBJECT))
-                                /* The session is already closed when we're querying the property */
-                                continue;
+                        log_full_errno(sd_bus_error_has_name(&e, SD_BUS_ERROR_UNKNOWN_OBJECT) ? LOG_DEBUG : LOG_WARNING,
+                                       r,
+                                       "Failed to get properties of session %s, ignoring: %s",
+                                       id, bus_error_message(&e, r));
+                        continue;
```
Comment (YHNdnzj) [inline src/login/loginctl.c]: Please remove the empty line.

**Pull request merged**